test/utils/test_jinja_chat_extension.py

```diff
@@ -0,0 +1,339 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+
+import pytest
+from jinja2 import TemplateSyntaxError
+from jinja2.sandbox import SandboxedEnvironment
+
+from haystack_experimental.dataclasses.chat_message import ImageContent, ToolCall, ToolCallResult
+from haystack_experimental.utils.jinja_chat_extension import ChatMessageExtension, for_template
+
+
+class TestChatMessageExtension:
+    @pytest.fixture
+    def jinja_env(self) -> SandboxedEnvironment:
+        # we use a SandboxedEnvironment here to replicate the conditions of the ChatPromptBuilder component
+        env = SandboxedEnvironment(extensions=[ChatMessageExtension])
+        env.filters["for_template"] = for_template
+        return env
+
+    def test_message_with_name_and_meta(self, jinja_env):
+        template = """
+        {% message role="user" name="Bob" meta={"language": "en"} %}
```

Comment:
Could we also add tests checking for the errors that occur when name and/or meta are malformed? E.g. if meta isn't valid JSON